configuration-reference/functions/firstwithtime.md

```diff
@@ -0,0 +1,24 @@
+---
+description: This section contains reference documentation for the firstwithtime function.
+---
+
+# firstwithtime
+
+Get the first value of `dataColumn` where the `timeColumn` is used to define the time of `dataColumn` and the `dataType` specifies the type of `dataColumn`, which can be `BOOLEAN`, `INT`, `LONG`, `FLOAT`, `DOUBLE`, `STRING`
+
+## Signature
+
+> FIRSTWITHTIME(dataColumn, timeColumn, 'dataType')
+
+## Usage Example
+
+This example is based on the [Streaming Quick Start](../../basics/getting-started/quick-start.md#streaming).
+
+```sql
+select FIRSTWITHTIME(group_name, __metadata$recordTimestamp, 'STRING')
```

Comment:
Looking at defined schema column used for timeExpression is actually a string 
    {
      "dataType": "STRING",
      "name": "__metadata$recordTimestamp"
    }
and not numeric per code definition, however I am guessing this query runs fine? If yes then perhaps internally we do implicit String to numeric conversion but then more relatable e.g could be just using time col. for the time `mTime`. Thoughts?